peps/pep-0705.rst

```diff
@@ -451,7 +483,35 @@ Notes:
     d: D = { "v": 1, "x": { "v": 1.1 } }
     a: A = deepcopy(d)  # Type check error: "v" is of type float in E
 
-* Type checkers may choose to reject deep copying between complex recursive types for algorithmic complexity reasons. However, the error should be explicit that this is a type checker limitation, and not due to an unsound assignment in the user code.
+* Type checkers may choose to reject deep copying between complex recursive types for algorithmic complexity reasons. However, the error should be explicit that this is a type checker limitation, and the user code is not necessarily unsound.
+
+Minimum support
+^^^^^^^^^^^^^^^
+
+A type checker may reject sound assignments, for instance if it is unable to support bidirectional type inference. This section provides a minimum standard of support that a type checker should provide, so users can write code that will be consistently accepted across all type checkers.
+
+Type checkers may reject any code not conforming to this model, but the error should be explicit that this is a type checker limitation, and that the user code is not necessarily unsound. It may be reasonable to use a distinct code for suppressing such issues, so if the checker later supports full inference and the code is indeed unsound, the issue will no longer be suppressed.
+
+A type checker may treat a shallow copy of a TypedDict ``A`` as returning a fixed TypedDict ``F`` that has the same items as ``A``, but with any read-only constraints removed.
+
+A type checker may treat a deep copy of a TypedDict ``A`` as returning a fixed TypedDict ``F`` constructed by the following rules:
+
+* The key set of ``F`` is the same as the key set of ``A``
+* All items in ``F`` are non-read-only
+* Keys in ``F`` are required if and only if they are required in ``A``
+* Value types are constructed based on the associated value type ``V`` in ``A`` according to the following rules:
+
+  * If ``V`` has already been seen, use the same associated type; in particular, recursive types should map to recursive types
+  * If ``V`` is a TypedDict and the item in ``A`` is read-only, apply these rules recursively
+  * Otherwise, use ``V``
+
+All type checkers should accept code that would be accepted if these rules were followed, even if the type checker does not internally construct such a type. However, if due to other limitations, a type checker would not consider ``F`` consistent with ``A`` (for instance due to incomplete support for recursive types), it should prioritize consistency.
+
+Notes:
+
+* For any TypedDict ``D`` with non-read-only items, the type constructed when creating a shallow or deep copy of ``D`` will be equivalent to ``D``, and the type checker may reuse ``D`` internally.
```

Comment:
Here again, I recommend deleting "and the type checker many reuse..." because this is an internal implementation detail.